frontend/src/components/Farm.js

```diff
@@ -1,61 +1,81 @@
 import React, { useState, useEffect } from "react";
-import Snek from "./Snek";
+import Snek from "./Snek/";
 
 const Farm = ({ farm }) => {
-  const defaultValues = {
+  const initialState = {
     micropythons: 0,
     curlyboi_snektions: 0
   }
 
-  const [state, setState] = useState(defaultValues)
+  const [state, setState] = useState(initialState)
   const {name, micropythons, curlyboi_snektions} = state
 
   useEffect(() => {
     farm ? setState(farm) : null
     }, []
   )
 
-  console.log("Data", state);
+  // Log out the state when it changes
+  useEffect(() => {
+    console.log("Updated State:", state)
+    }, [state]
+  )
+
   const updateSneks = () => {
     let micros = state.micropythons;
     let boi = state.curlyboi_snektions;
-      if (micros < 10 ) {
-        micros += 1;
-      } else  {
-        micros = 0;
-        boi += 1;
-      }
-      setState(
-        { ...state, micropythons: micros, curlyboi_snektions: boi }
-      )
-    console.log("New state: ", state)
-  }
 
-  console.log("Data", state);
-  let micropython_array = [];
-  let curlyboi = null;
-  for (var i = 0; i < micropythons; i++) {
-    micropython_array.push(<Snek size={1} length={5} />)
+    if (micros < 10 ) {
+      micros += 1;
+    } else  {
+      micros = 0;
+      boi += 1;
+    }
+
+    setState(
+      { ...state, micropythons: micros, curlyboi_snektions: boi }
+    )
   }
 
-  if(state.curlyboi_snektions > 0) {
-    curlyboi = <Snek size={3} length={state.curlyboi_snektions} />
+  const Micropythons = () => {
+    let micropys = [];
+    for (var i = 0; i < micropythons; i++) {
+      micropys.push(<Snek key={i} size="medium" length={5} />)
+    }
+    return(
+      <div className="micropythons">
+        {micropys}
+      </div>
+    )
   }
 
+  const Curlyboi = () => (
+    state.curlyboi_snektions > 0
+    ? <Snek size="big" length={state.curlyboi_snektions} />
+    : null
+  )
+
   return(
     <div>
-      <div>
-        <h1>Welcome to {name} farm</h1>
-        <p>You have collected {micropythons} micropythons</p>
-        <button onClick={updateSneks}>Moar sneks, please!</button>
-      </div>
-      <h2>Micropythons</h2>
-      <div className="micropythons">
-        { micropython_array }
+      <header>
+        <h1>Welcome to {name} Farm</h1>
+      </header>
+
+      <div className="farm farm--flexContainer">
+        <section className="farm farm--buttons">
+          <button onClick={updateSneks}>Moar sneks, please!</button>
+        </section>
+
+        <section className="farm farm--longGrass">
+          <Micropythons/>
+          <Curlyboi/>
+        </section>
+
+        <section className="farm farm--snekBoard">
```

Comment:
Honestly, I don't know CSS Naming and BEM very well, and was just giving it my best shot 😂 I'll take a look at the BEM Specifications and give this another go! I'm mostly fumbling around in the dark when it comes to CSS - most my experience is with CSS-in-JS which is a different beast, and doesn't use as much naming.

I think my overall naming structure was to try to have things labelled as "farm" since they're sub-elements of the "farm" container, but perhaps I went overboard with the `--`s, and also, now that I think of it, giving them all the class `farm` kind of defeats the point, when I can instead target "all farm elements" using CSS Selectors etc. I'll rethink this